### ISSUE TITLE
Top-level await behavior potentially different than native implementations?

### ISSUE THREAD

**Issue by evanw** [systemjs/systemjs#2295]
### Demonstration

I'm looking at adding TLA to [esbuild](https://github.com/evanw/esbuild) and I'm trying to get the behavior of the bundled code to match the unbundled code. So I'm investigating various existing approaches to bundling TLA. One of them is to use a `System.register` call for each module and put all of the calls in the bundle file, then call `System.import` on the entry point. I have written a fuzzer that includes this bundling strategy: https://github.com/evanw/tla-fuzzer.

### Expected Behavior

I expected bundling with SystemJS to match the behavior of the unbundled code because I was under the impression that SystemJS is designed to match the behavior of a native implementation.

### Actual Behavior

The evaluation order of SystemJS appears to be different from V8 according to my fuzzer. I'm creating this issue here because I'm not sure if it's a problem with SystemJS, V8, or with my interpretation of the specification and I figured this could be a good place to ask. Do you know if this is a problem with SystemJS?

**Comment by guybedford** [comment]
Yes, supporting the proper execution chains of the spec is on my list for SystemJS, and it is not supported currently. The current implementation is the naive parallel Promise.all() for dependencies which will result in the extra ticks between sync parents and async children.

SystemJS can support the exact semantics just fine though, it just requires reworking the execution algorithm. Basically following the execution specification exactly would be the thing to do for SystemJS here. I was waiting for implementation pressure to drive this work, so thank you for bringing it up!

Let's keep this issue open as a tracking issue for the alignment work.

**Comment by jogelin** [comment]
Any updates on this? (some pressure :D)
